app/scripts/migrations/093.test.ts

```diff
@@ -0,0 +1,212 @@
+          ticker: 'ETH',
+          type: InfuraNetworkType.goerli,
+          chainId: '5',
+          nickname: 'Goerli Testnet',
+        },
+      },
+    };
+    const oldStorage = {
+      meta: {
+        version: 92,
+      },
+      data: oldData,
+    };
+
+    const newStorage = await migrate(oldStorage);
+    expect(newStorage.data).toStrictEqual(oldData);
+  });
+
+  it('should migrate networkStatus into networksMetadata keyed by the active providerConfig.id value, calling the new value "status" and deleting the root networkStatus value', async () => {
+    const oldData = {
+      other: 'data',
+      NetworkController: {
+        providerConfig: {
+          type: NetworkType.rpc,
+          chainId: '0x9393',
+          nickname: 'Funky Town Chain',
+          ticker: 'ETH',
+          id: 'test-network-client-id',
+        },
+        networkStatus: NetworkStatus.Available,
+      },
+    };
+    const oldStorage = {
+      meta: {
+        version: 92,
+      },
+      data: oldData,
+    };
+
+    const newStorage = await migrate(oldStorage);
+    expect(newStorage.data).toStrictEqual({
+      other: 'data',
+      NetworkController: {
+        providerConfig: {
+          type: NetworkType.rpc,
+          chainId: '0x9393',
+          nickname: 'Funky Town Chain',
+          ticker: 'ETH',
+          id: 'test-network-client-id',
+        },
+        networksMetadata: {
+          'test-network-client-id': {
+            status: NetworkStatus.Available,
+            EIPS: {},
+          },
+        },
+      },
+    });
+  });
+
+  it('should migrate the contents of networkDetails into networksMetadata keyed by the active providerConfig.id value, and delete the root networkDetails value', async () => {
+    const oldData = {
+      other: 'data',
+      NetworkController: {
+        providerConfig: {
+          type: NetworkType.rpc,
+          chainId: '0x9393',
+          nickname: 'Funky Town Chain',
+          ticker: 'ETH',
+          id: 'test-network-client-id',
+        },
+        networkDetails: {
+          EIPS: { 1559: false },
+        },
+      },
+    };
+    const oldStorage = {
+      meta: {
+        version: 92,
+      },
+      data: oldData,
+    };
+
+    const newStorage = await migrate(oldStorage);
+    expect(newStorage.data).toStrictEqual({
+      other: 'data',
+      NetworkController: {
+        providerConfig: {
+          type: NetworkType.rpc,
+          chainId: '0x9393',
+          nickname: 'Funky Town Chain',
+          ticker: 'ETH',
+          id: 'test-network-client-id',
+        },
+        networksMetadata: {
+          'test-network-client-id': {
+            status: NetworkStatus.Unknown,
+            EIPS: {
+              1559: false,
+            },
+          },
+        },
+      },
+    });
+  });
+
+  it('should update state migrating networkStatus and networkDetails state into networksMetadata keyed by the active providerConfig.id value', async () => {
```

Comment:
should this be split into two separate tests?

Reply:
the two tests directly above do kindof split it out and then this one does a combined one for good measure.

Reply:
my bad!